lib/widgets/app.dart

```diff
@@ -57,6 +57,13 @@ class HomePage extends StatelessWidget {
               child: Column(children: [
                 const Text('🚧 Under construction 🚧'),
                 const SizedBox(height: 12),
+                ListenableBuilder(
+                  builder: (BuildContext context, Widget? child) {
+                    return Text.rich(TextSpan(
+                      text: 'You are: ',
+                      children: [bold(store.full_name)]));},
+                  listenable: store,
+                ),
```

Comment:
Neat!

Is this `ListenableBuilder` necessary, though? This `HomePage.build` method already calls `PerAccountStoreWidget.of`, which in turn calls `context.dependOnInheritedWidgetOfExactType<PerAccountStoreWidget>()`. That "dependOn…" method should cause `HomePage` itself to get rebuilt after the store notifies its listeners. So if that wasn't working in your testing, we should investigate (perhaps in chat.)

Formatting nit: generally the "child", "children", "builder", or similar argument that provides the nested bit of UI goes last. So:

```suggestion
                ListenableBuilder(
                  listenable: store,
                  builder: (BuildContext context, Widget? child) {
                    return Text.rich(TextSpan(
                      text: 'You are: ',
                      children: [bold(store.full_name)]));
                  }),
```

Reply:
I've also just written some docs for that `PerAccountStoreWidget.of` method: https://github.com/zulip/zulip-flutter/commit/8a5fb4d6b86d9f23fdf93f6168e83df192eca77b in #26. Take a look and see if that helps clarify things.

Reply:
I explored various ways of notifying the UI (eg. notifying on individual `String`s), before noticing that the whole account was a notifier via inheritance, and that I needed to explicitly `notifyListeners()` - which possibly is not necessary with the simple values eg. `String`.

(I'd not reached the point of exploring trimming out the builder itself beyond that; it was more that things were working so wanted to push before spending too much extra time on it, in case going in completely the wrong direction)

I hadn't dug into that `.of` call. That Widget is intended as a way to easily associate an account with another widget class, and auto-build on data-change?

Re the builder being necessary, apparently it works without it :+1:

I'm not sure if they can be optimization to include it inline, if the base class widget always triggers a rebuild?

I can understand the nit of preferring the smaller `listenable:` first. I was going from the flutter examples of this, which have them last.

Reply:
> Re the builder being necessary, apparently it works without it +1

Great. :-)  If it hadn't, that would have pointed to something we should definitely investigate to revise my understanding of how this works — and though I was pretty sure it *should* work from reading the docs, we hadn't really exercised it and so I wasn't sure if it would turn out I'd been missing something.

> I explored various ways of notifying the UI (eg. notifying on individual `String`s), before noticing that the whole account was a notifier via inheritance, and that I needed to explicitly `notifyListeners()` -

Yeah, that's what I hoped had happened.

I'll start a chat thread for replying to the other questions.